ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/rewrite/testtool/templates/Divulgence2.scala

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.rewrite.testtool.templates
+
+import com.daml.ledger.api.rewrite.testtool.infrastructure.LedgerTestContext
+import com.digitalasset.ledger.api.v1.value.Value.Sum.{ContractId, Party}
+
+import scala.concurrent.Future
+
+object Divulgence2 {
+  def apply(party: String, div2Signatory: String, div2Fetcher: String)(
+      implicit context: LedgerTestContext): Future[Divulgence2] =
+    context
+      .create(
+        party,
+        ids.divulgence2,
+        Map("div2Signatory" -> new Party(div2Signatory), "div2Fetcher" -> new Party(div2Fetcher))
+      )
+      .map(new Divulgence2(_, party, div2Signatory, div2Fetcher) {})
+
+}
+
+sealed abstract case class Divulgence2(
```

Comment:
Why is it an abstract case class?

Reply:
Controlling the constructor while letting the compiler synthesize the extractors ([reference](https://gist.github.com/tpolecat/a5cb0dc9adeacc93f846835ed21c92d2)).

Reply:
Thanks for the reference 👍